### PR TITLE
Feature/sub sector expansion

### PR DESCRIPTION
**Pull request by tnagorra**
Depends on
https://github.com/toggle-corp/react-store/pull/632
https://github.com/toggle-corp/react-store/pull/674

### CODE REVIEW

File: src/components/general/OrientationInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        </div>
+    ) : data.label
+);
+
+
+export default class OrientationInput extends React.PureComponent<Props, State> {
+    render() {
+        const {
+            // FIXME: string
+            label = 'Orientation',
```
Comment (tnagorra) [inline src/components/general/OrientationInput/index.tsx:44]: Please use string here!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 2]
```diff
@@ -61,6 +61,27 @@ export default class WidgetEditor extends React.PureComponent {
     static schema = {};
     static value = {};
 
+    /*
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We should remove commented code!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 3]
```diff
@@ -74,17 +95,39 @@ export default class WidgetEditor extends React.PureComponent {
         return minSize;
     }
 
-    handleItemChange = (newWidget) => {
+    /*
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We should remove commented code!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 4]
```diff
@@ -132,57 +175,75 @@ export default class WidgetEditor extends React.PureComponent {
...
+        const headingClassName = _cs(
+            styles.heading,
+            hideButtons && styles.disabled,
+        );
 
-        const headingClassName = `
-            ${styles.heading}
-            ${hideButtons ? styles.disabled : ''}
-        `;
+        // const { editComponent: Widget } = fetchWidget(widgetType, widgetId);
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We should remove commented code!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 5]
```diff
@@ -132,57 +175,75 @@ export default class WidgetEditor extends React.PureComponent {
...
+                        <Fragment>
+                            <WarningButton
+                                iconName="edit"
+                                title={_ts('framework.widgetEditor', 'editTooltip')}
+                                tabIndex="-1"
+                                transparent
+                                onClick={() => this.props.onWidgetEditClick(key, widget)}
+                                disabled={widgetsDisabled}
+                                // TODO: add onClick handler
+                                /*
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We should remove commented code!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 6]
```diff
@@ -200,38 +261,33 @@ export default class WidgetEditor extends React.PureComponent {
...
-        const className = `
-            ${styles.content}
-            ${isDisabled ? styles.disabled : ''}
-        `;
+        const className = _cs(
+            styles.content,
+            isDisabled && styles.disabled,
+        );
+
+        /*
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We should remove commented code!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 7]
```diff
@@ -240,6 +296,7 @@ export default class WidgetEditor extends React.PureComponent {
                         <Widget
                             widgetName={widgetId}
                             widgetType={widgetType}
+                            // widget={modifiedWidget}
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We should remove commented code!

File: src/views/AnalysisFramework/WidgetEditor/index.js [hunk 8]
```diff
@@ -250,15 +307,10 @@ export default class WidgetEditor extends React.PureComponent {
...
-                { isDisabled && (
-                    <div
-                        className={styles.disablerMask}
-                        style={{
-                            fontSize: `${fontSize}px`,
-                        }}
-                    >
-                        { disablerMaskText }
-                    </div>
+                { isDisabled && ( // FIXME: use Message component?
```
Comment (tnagorra) [inline src/views/AnalysisFramework/WidgetEditor/index.js]: We can remove the FIXME as Message component is already used

File: src/views/DiscoverProjects/headers.js [hunk 9]
```diff
@@ -45,6 +45,7 @@ const headers = [
         key: 'number_of_users',
         title: _ts('discoverProjects.table', 'numberOfUsersTitle'),
         order: 8,
+        sortable: true,
```
Comment (AdityaKhatri) [inline src/views/DiscoverProjects/headers.js]: To be reomved

File: src/views/EditEntries/Overview/index.js [hunk 10]
```diff
@@ -301,7 +301,9 @@ export default class Overview extends React.PureComponent {
                         <WidgetFaram
                             className={styles.content}
                             // NOTE: dismount on key change
-                            key={key}
+                            // NOTE: removed dismount on key change behavior
+                            // to persist active UI state
+                            // key={key}
```
Comment (tnagorra) [inline src/views/EditEntries/Overview/index.js:306]: We should think about this

File: src/widgetComponents/Matrix2dInput/DimensionRow/SubdimensionRow/index.js [hunk 11]
```diff
@@ -25,16 +27,88 @@ export default class SubdimensionRow extends React.PureComponent {
...
+            subdimensionId,
+            sectorId,
+            activeSectorKey,
+        } = this.props;
+
+        if (!activeSectorKey) {
+            return false;
+        }
+
+        const subsectors = value && ((value[dimensionId] || {})[subdimensionId]);
```
Comment (tnagorra) [inline src/widgetComponents/Matrix2dInput/DimensionRow/SubdimensionRow/index.js]: Lets use object de-structuring here

File: src/widgetComponents/Matrix2dInput/DimensionRow/index.js [hunk 12]
```diff
@@ -37,17 +39,20 @@ export default class DimensionRow extends React.PureComponent {
...
                 getRgbRawFromHex(rowStyle.color),
-                0.4,
+                0.3,
             ),
         );
 
+        const textBorderColor = firstColor;
         return {
+            textShadow: `-1px 0 ${textBorderColor}, 0 1px ${textBorderColor}, 1px 0 ${textBorderColor}, 0 -1px ${textBorderColor}`,
+            // textShadow: '0 0 red'
```
Comment (tnagorra) [inline src/widgetComponents/Matrix2dInput/DimensionRow/index.js]: We shouldn't add un-necessary comments

File: src/views/AnalysisFramework/List/index.js [hunk 13]
```diff
@@ -17,25 +18,67 @@ const listWidgets = widgetList.filter(
...
+            onWidgetSave,
+            onWidgetChange,
+            onWidgetCancel,
+        } = this.props;
+        return (
+            <div className={styles.widgetList}>
+                <Widget
+                    widgetKey={widget.key}
+                    title={widget.title}
+                    data={widget.properties.data}
```
Comment (AdityaKhatri) [inline src/views/AnalysisFramework/List/index.js:60]: @tnagorra don't we need to check for undefined?

File: src/widgetComponents/Matrix2dInput/index.js [hunk 14]
```diff
@@ -50,43 +34,92 @@ export default class Matrix2dInput extends React.PureComponent {
...
+        });
+
         onChange(newValue);
     }
 
-    handleCellDrop = (dimensionId, subdimensionId, sectorId, droppedData) => {
+    handleCellDrop = (dimensionId, subdimensionId, sectorId, subsectorId, droppedData) => {
         const { type, data } = droppedData;
 
+        console.warn('Sector', sectorId, 'Subsector', subsectorId);
```
Comment (tnagorra) [inline src/widgetComponents/Matrix2dInput/index.js]: Lets not add un-necesary console.warn

File: src/widgets/edit/Matrix2d/Column/index.tsx [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+const emptyList: unknown[] = [];
+const emptyObject = {};
+
+export default class Column extends React.PureComponent<Props, State> {
+    private sectorItemRendererParams = (key, elem, i) => ({
+        className: styles.sectorContent,
+        data: elem,
+        faramElementName: String(i),
+        index: i,
+        onEditButtonClick: () => {
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/Column/index.tsx]: We shouldn't create new function inside rendererParams

File: src/widgets/edit/Matrix2d/Column/index.tsx [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+            onLinkWidgetModalVisiblityChange,
+            onSectorContentBackButtonClick,
+            selectedSectorKey,
+            titleSelector,
+            widgetKey,
+        } = this.props;
+
+        const { sectors = emptyList } = faramValues;
+        const selectedSectorIndex = sectors
+            .findIndex(sector => (keySelector(sector) === selectedSectorKey));
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/Column/index.tsx]: Lets memoize this

File: src/widgets/edit/Matrix2d/DimensionContent/SubdimensionRow/index.js [hunk 17]
```diff
@@ -15,20 +19,48 @@ const deleteClick = (options, index) => {
...
+                    />
+                    <OrientationInput
+                        className={styles.orientationInput}
+                        faramElementName="orientation"
+                        persistantHintAndError={false}
+                    />
+                    <TextInput
+                        type="number"
+                        // FIXME: string
+                        label="Font size"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/DimensionContent/SubdimensionRow/index.js]: Use string

File: src/widgets/edit/Matrix2d/DimensionContent/SubdimensionRow/index.js [hunk 18]
```diff
@@ -15,20 +19,48 @@ const deleteClick = (options, index) => {
...
+                        className={styles.fontSizeInput}
+                        faramElementName="fontSize"
+                        persistantHintAndError={false}
+                    />
+                    <TextInput
+                        type="number"
+                        // FIXME: string
+                        label="Height"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/DimensionContent/SubdimensionRow/index.js]: Use string

File: src/widgets/edit/Matrix2d/DimensionContent/index.js [hunk 19]
```diff
@@ -59,91 +77,132 @@ export default class DimensionContent extends React.PureComponent {
...
+                                persistantHintAndError={false}
+                            />
+                            <OrientationInput
+                                className={styles.orientationInput}
+                                faramElementName="orientation"
+                                persistantHintAndError={false}
+                            />
+                            <TextInput
+                                type="number"
+                                label="Font size"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/DimensionContent/index.js]: use string

File: src/widgets/edit/Matrix2d/DimensionContent/index.js [hunk 20]
```diff
@@ -59,91 +77,132 @@ export default class DimensionContent extends React.PureComponent {
...
+                                className={styles.fontSizeInput}
+                                faramElementName="fontSize"
+                                persistantHintAndError={false}
+                            />
+                            <TextInput
+                                type="number"
+                                label="Height"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/DimensionContent/index.js]: Use string

File: src/widgets/edit/Matrix2d/Row/index.tsx [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+const emptyObject = {};
+
+export default class Row extends React.PureComponent<Props, State> {
+    private dimensionItemRendererParams = (key, elem, i) => ({
+        className: styles.dimensionContent,
+        data: elem,
+        faramElementName: String(i),
+        index: i,
+        keySelector: this.props.keySelector,
+        onEditButtonClick: () => {
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/Row/index.tsx]: We shouldn't create new function inside rendererParams

File: src/widgets/edit/Matrix2d/Row/index.tsx [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+            onLinkWidgetModalVisiblityChange,
+            onDimensionContentBackButtonClick,
+            selectedDimensionKey,
+            titleSelector,
+            widgetKey,
+        } = this.props;
+
+        const { dimensions = emptyList } = faramValues;
+        const selectedDimensionIndex = dimensions.findIndex(
+            dimension => (keySelector(dimension) === selectedDimensionKey),
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/Row/index.tsx]: Lets memoize this

File: src/widgets/edit/Matrix2d/SectorContent/SubsectorRow/index.js [hunk 23]
```diff
@@ -15,20 +19,47 @@ const deleteClick = (options, index) => {
...
+                    />
+                    <OrientationInput
+                        className={styles.orientationInput}
+                        faramElementName="orientation"
+                        persistantHintAndError={false}
+                    />
+                    <TextInput
+                        type="number"
+                        // FIXME: string
+                        label="Font size"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/SectorContent/SubsectorRow/index.js]: Use string

File: src/widgets/edit/Matrix2d/SectorContent/SubsectorRow/index.js [hunk 24]
```diff
@@ -15,20 +19,47 @@ const deleteClick = (options, index) => {
...
+                        className={styles.fontSizeInput}
+                        faramElementName="fontSize"
+                    />
+                    <TextInput
+                        type="number"
+                        // FIXME: string
+                        label="Width"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/SectorContent/SubsectorRow/index.js]: Use string

File: src/widgets/edit/Matrix2d/SectorContent/index.js [hunk 25]
```diff
@@ -58,27 +77,63 @@ export default class SectorContent extends React.PureComponent {
...
+                            />
+                            <OrientationInput
+                                className={styles.orientationInput}
+                                faramElementName="orientation"
+                                persistantHintAndError={false}
+                            />
+                            <TextInput
+                                type="number"
+                                // FIXME: string
+                                label="Font size"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/SectorContent/index.js]: Use string

File: src/widgets/edit/Matrix2d/SectorContent/index.js [hunk 26]
```diff
@@ -58,27 +77,63 @@ export default class SectorContent extends React.PureComponent {
...
+                                className={styles.fontSizeInput}
+                                faramElementName="fontSize"
+                            />
+                            <TextInput
+                                type="number"
+                                // FIXME: string
+                                label="Width"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/SectorContent/index.js]: Use string

File: src/widgets/edit/Matrix2d/index.js [hunk 27]
```diff
@@ -402,196 +407,120 @@ export default class Matrix2dEditWidget extends React.PureComponent {
...
+                        placeholder={_ts('widgets.editor.matrix2d', 'widgetTitlePlaceholder')}
+                        selectOnFocus
+                        persistantHintAndError={false}
+                    />
+                    <div className={styles.metaInputs}>
+                        <FaramGroup faramElementName="meta">
+                            <TextInput
+                                className={styles.titleRowHeightInput}
+                                faramElementName="titleRowHeight"
+                                label="Title row height"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/index.js]: Use string

File: src/widgets/edit/Matrix2d/index.js [hunk 28]
```diff
@@ -402,196 +407,120 @@ export default class Matrix2dEditWidget extends React.PureComponent {
...
+                                type="number"
+                                persistantHintAndError={false}
+                            />
+                            <TextInput
+                                type="number"
+                                label="Title column width"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/index.js]: Use string

File: src/widgets/edit/Matrix2d/index.js [hunk 29]
```diff
@@ -402,196 +407,120 @@ export default class Matrix2dEditWidget extends React.PureComponent {
...
+                                className={styles.titleColumnWidthInput}
+                                faramElementName="titleColumnWidth"
+                                persistantHintAndError={false}
+                            />
+                            <TextInput
+                                type="number"
+                                label="Subtitle column width"
```
Comment (tnagorra) [inline src/widgets/edit/Matrix2d/index.js]: Use string

File: src/widgets/edit/Matrix2d/index.js [hunk 30]
```diff
@@ -154,6 +164,11 @@ export default class Matrix2dEditWidget extends React.PureComponent {
                                     id: [requiredCondition],
                                     tooltip: [],
                                     title: [requiredCondition],
+
+                                    // TODO: implement orientation and fontSize
```
Comment (AdityaKhatri) [inline src/widgets/edit/Matrix2d/index.js]: Isn't this already implemented?

**Pull request merged**